src/sync/sync_manager.cpp

```diff
@@ -411,6 +411,33 @@ void SyncManager::set_current_user(const std::string& user_id)
     m_metadata_manager->set_current_user_identity(user_id);
 }
 
+void SyncManager::remove_user(const std::string& user_id)
+{
+    std::lock_guard<std::mutex> lock(m_user_mutex);
+
+    auto it = std::find_if(m_users.begin(),
+                           m_users.end(),
+                           [user_id](const auto& user) {
+        return user->identity() == user_id;
+    });
+    
+    if (it == m_users.end())
+        return;
+    
+    auto user = *it;
+    
+    if (!m_metadata_manager) {
+        return;
+    }
+    
+    if (user->identity() == m_metadata_manager->get_current_user_identity()) {
+        m_metadata_manager->set_current_user_identity("");
+    }
+    
+    user->set_state(SyncUser::State::Removed);
+    
```

Comment:
[major] You also need to mark the user for removal. This is not fully removing the user.

Reply:
done